### ISSUE TITLE
Use schemas for domain states too

### ISSUE THREAD

**Issue by Kazuo256** [uspgamedev/backdoor#242]
In ordor to automatically derive the devmode editors of in-game object states.

**Comment by OrenjiAkira** [comment]
explain, what do you mean by states? the way we save them? since they have some depth isn't that a little overcomplicated?

**Comment by Kazuo256** [comment]
Yeah, now that you mention it, the depth is a problem. What I wanted was simply to map what variables there are in the state of elements (the damage in bodies, the cooldown in actors, the tiles in sectors, etc.) so we didn't have to manually add input fields in the devmode code.

Given that, I can think of three options:
1. The schemas for states don't have to map ALL variables, jsut the ones we want to edit in devmode. This way we can leave `sector.bodies` out of it, for instance.
2. Swap all references to elements for their respective IDs, "flattening" the state.
3. Give up and write it all manually XD

**Comment by OrenjiAkira** [comment]
I'm pretty sure this is obsolete, right?

**Comment by Kazuo256** [comment]
No it is not. It would still be quite useful. Basically a "save file" editor in-game made easy.
